File: src/types/table.ts
export type TableShape  = "square" | "round" | "rect";
export type TableStatus = "libre" | "occupe" | "addition";

export interface Room {
  id: string;
  name: string;
  color: string | null;
  sort_order: number;
}

export interface RestaurantTable {
  id: string;
  room_id: string | null;
  name: string;
  seats: number;
  shape: TableShape;
  status: TableStatus;
  pos_x: number;
  pos_y: number;
  sort_order: number;
}

File: src-tauri/src/db/models/table.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Room {
    pub id: String,
    pub name: String,
    pub color: Option<String>,
    pub sort_order: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct RestaurantTable {
    pub id: String,
    pub room_id: Option<String>,
    pub name: String,
    pub seats: i64,
    pub shape: String,
    pub status: String,
    pub pos_x: i64,
    pub pos_y: i64,
    pub sort_order: i64,
}
